bin/contracts.ml

```diff
@@ -0,0 +1,44 @@
+(** Foundational definitions *)
+type currency = USD | GBP | JPY | SGD;;
+type day = Int;;
+type month = Int;;
+type year = Int;;
+type date = Date of year * month * day;;
```

Comment:
Might want to see if Ocaml has a library that does like Date or something

Reply:
next pr :)